### ISSUE TITLE
`level` is identified as a Snowflake keyword, but it isn't 

### ISSUE THREAD

**Issue by foundinblank** [sqlfluff/sqlfluff#1823]
## Expected Behaviour

Field aliased or named `level` should pass, since it's not a [Snowflake reserved keyword](https://docs.snowflake.com/en/sql-reference/reserved-keywords.html). 

## Observed Behaviour

sqlfluff lint fails since it claims `level` is a keyword which triggers L029: "Keywords should not be used as identifiers." 

## Steps to Reproduce

```sql
select
    col1 as col,
    col2 as level
from schema.tbl
```

Output of sqlfluff lint: 
```
❯ sqlfluff lint models/example.sql
=== [dbt templater] Sorting Nodes...
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
== [models/example.sql] FAIL
L:   3 | P:  13 | L029 | Keywords should not be used as identifiers.
All Finished 📜 🎉!
```

Output of sqlfluff parse:
```
❯ sqlfluff parse models/example.sql
=== [dbt templater] Compiling dbt project...
=== [dbt templater] Project Compiled.
[L:  1, P:  1]      |file:
[L:  1, P:  1]      |    statement:
[L:  1, P:  1]      |        select_statement:
[L:  1, P:  1]      |            select_clause:
[L:  1, P:  1]      |                keyword:                                      'select'
[L:  1, P:  7]      |                [META] indent:
[L:  1, P:  7]      |                newline:                                      '\n'
[L:  2, P:  1]      |                whitespace:                                   '    '
[L:  2, P:  5]      |                select_clause_element:
[L:  2, P:  5]      |                    column_reference:
[L:  2, P:  5]      |                        identifier:                           'col1'
[L:  2, P:  9]      |                    whitespace:                               ' '
[L:  2, P: 10]      |                    alias_expression:
[L:  2, P: 10]      |                        keyword:                              'as'
[L:  2, P: 12]      |                        whitespace:                           ' '
[L:  2, P: 13]      |                        identifier:                           'col'
[L:  2, P: 16]      |                comma:                                        ','
[L:  2, P: 17]      |                newline:                                      '\n'
[L:  3, P:  1]      |                whitespace:                                   '    '
[L:  3, P:  5]      |                select_clause_element:
[L:  3, P:  5]      |                    column_reference:
[L:  3, P:  5]      |                        identifier:                           'col2'
[L:  3, P:  9]      |                    whitespace:                               ' '
[L:  3, P: 10]      |                    alias_expression:
[L:  3, P: 10]      |                        keyword:                              'as'
[L:  3, P: 12]      |                        whitespace:                           ' '
[L:  3, P: 13]      |                        identifier:                           'level'
[L:  3, P: 18]      |            newline:                                          '\n'
[L:  4, P:  1]      |            [META] dedent:
[L:  4, P:  1]      |            from_clause:
[L:  4, P:  1]      |                keyword:                                      'from'
[L:  4, P:  5]      |                whitespace:                                   ' '
[L:  4, P:  6]      |                from_expression:
[L:  4, P:  6]      |                    [META] indent:
[L:  4, P:  6]      |                    from_expression_element:
[L:  4, P:  6]      |                        table_expression:
[L:  4, P:  6]      |                            table_reference:
[L:  4, P:  6]      |                                identifier:                   'schema'
[L:  4, P: 12]      |                                dot:                          '.'
[L:  4, P: 13]      |                                identifier:                   'tbl'
[L:  4, P: 16]      |                    [META] dedent:
[L:  4, P: 16]      |    newline:                                                  '\n'
```

## Dialect

`snowflake`

## Version
sqlfluff 0.7.0
python 3.7.9

## Configuration
```
[sqlfluff]
verbose = 0
nocolor = False
dialect = snowflake
templater = dbt
```

**Comment by tunetheweb** [comment]
This was already fixed in #1737 , which will be in the next release - which we planning to release in next few days.

**Comment by foundinblank** [comment]
That's the best kind of response to get on a new PR, thanks!

**Comment by tunetheweb** [comment]
Might be worth checking out the `main` branch to ensure it's definitely fixed for you. And that the keyword clean up hasn't introduced any new problems for you!
